### PR TITLE
M48:Enable simd.js in v8-crosswalk next

### PR DESCRIPTION
**Pull request by chuan9**
The implementation is only for ia32 and x64 ports.
Inline all SIMD operations and emit SIMD instructions for ia32 and x64 port
BUG=XWALK-5873

### CODE REVIEW

**Comment by chuan9** [comment]
@huningxin Please have a look, thanks～

File: src/crankshaft/hydrogen.cc [hunk 1]
```diff
@@ -6532,9 +6556,41 @@ bool HOptimizedGraphBuilder::PropertyAccessInfo::NeedsWrappingFor(
 }
 
 
+/*
```
Comment (huningxin) [inline src/crankshaft/hydrogen.cc]: Please remove the commented code.

File: src/crankshaft/hydrogen.cc [hunk 2]
```diff
@@ -6573,6 +6629,20 @@ HValue* HOptimizedGraphBuilder::BuildMonomorphicAccess(
 
   if (info->IsData()) {
     if (info->IsLoad()) {
+/*
```
Comment (huningxin) [inline src/crankshaft/hydrogen.cc]: Ditto.

File: src/crankshaft/hydrogen.h [hunk 3]
```diff
@@ -2627,6 +2629,23 @@ class HOptimizedGraphBuilder : public HGraphBuilder, public AstVisitor {
     }
     bool IsConfigurable() const { return details_.IsConfigurable(); }
     bool IsReadOnly() const { return details_.IsReadOnly(); }
+    bool IsSIMD128PropertyCallback() {
+/*
```
Comment (huningxin) [inline src/crankshaft/hydrogen.h]: Do we really need this API? Please remove it if it is not used.

File: src/crankshaft/ia32/lithium-ia32.cc [hunk 4]
```diff
@@ -351,23 +351,41 @@ void LAccessArgumentsAt::PrintDataTo(StringStream* stream) {
...
+      spill_slot_count_++;
+      spill_slot_count_ |= 1;
+      num_double_slots_++;
+      return spill_slot_count_++;
+    }
+    case FLOAT32x4_REGISTERS:
+    case BOOL32x4_REGISTERS:
+    case INT32x4_REGISTERS: {
+      // Skip three slots if for a quad-width slot.
+      spill_slot_count_ += 3;
```
Comment (huningxin) [inline src/crankshaft/ia32/lithium-ia32.cc]: What does make the code different between SIMD registers and double registers?

File: src/ia32/macro-assembler-ia32.cc [hunk 5]
```diff
@@ -1518,6 +1518,28 @@ void MacroAssembler::AllocateHeapNumber(Register result,
...
+  V(Float32x4, float32x4)                  \
+  V(Int32x4, int32x4)                      \
+  V(Bool32x4, bool32x4)
+
+#define DECLARE_SIMD_HEAP_ALLOCATE_FUNCTION(Type, type)                  \
+void MacroAssembler::Allocate##Type(Register result,                     \
+                                    Register scratch1,                   \
+                                    Register scratch2,                   \
+                                    Label* gc_required) {                \
+  /* Allocate SIMD128 object */                                                \
```
Comment (huningxin) [inline src/ia32/macro-assembler-ia32.cc]: Please keep the "\" aligned.

File: src/crankshaft/ia32/lithium-ia32.cc (hunk 4 above)
Comment (chuan9) [inline src/crankshaft/ia32/lithium-ia32.cc]: @huningxin This was cherried pick from previous implementation. Maybe one simd128 register need two double registers and every double register consume two spill slots on ia32 arch. So two double registers consume 4 spill slots.

**Comment by chuan9** [comment]
@huningxin Resubmit this patch, please have a look.

File: src/crankshaft/ia32/lithium-ia32.cc (hunk 4 above)
Comment (huningxin) [inline src/crankshaft/ia32/lithium-ia32.cc]: Yup. That's the reason.

**Comment by huningxin** [comment]
Code LGTM. Please make sure the code passes the coding style checker and all unit test on our targeted platforms ia32, x64, arm and arm64. Thanks.

**Comment by chuan9** [comment]
@huningxin yeah, I've done these work locally. Thanks for review!

**Comment by rakuco** [comment]
@chuan9 can you just update the commit message and remove the "Changes to be committed" part?

**Comment by chuan9** [comment]
@rakuco Done! Thanks!

**Comment by rakuco** [comment]
Thanks!

**Pull request merged**